### DISCUSSION TITLE
Class fields support without custom parser?

### DISCUSSION THREAD

**Discussion by proko** [eslint/eslint#18461] [Q&A] [answered]
So, do we still need to use babel parser to get support for javascript class fields? Using v9.2.0 and getting error for `Unexpected token =`. 
This is my basic config file. Project on Vite 
```
export default [
  js.configs.recommended,
  {
    languageOptions: {
      ecmaVersion: "latest",
      sourceType: "module",
      globals: {
        ...globals.browser,
      },
    },
    rules: {
      "no-unused-vars": 1,
      "no-unsafe-optional-chaining": 1,
      "no-prototype-builtins": 1,
      "no-extra-boolean-cast": 1,
    },
  },
  ...compat.config({
    extends: ["plugin:cypress/recommended"],
  }),
];

```

**Comment by fasttime** [comment]
@proko Class fields are fully supported by the default parser in ESLint v8 and v9, so the error you see is probably caused by something else. What does the file where the error occurs look like?

**Reply by proko** [reply]
hey @fasttime thanks for the comment. They are regular classes as far as I can say
```
import { action, makeObservable, observable } from "mobx";
import { createContext } from "react";

class UnregisterStore {
  unRegDocuments = [];
   ....

  constructor() {
    makeObservable(this, {
      unRegDocuments: observable,
      ...
    });
  }

  method = () => {
    ...
  }
}
export default createContext(new UnregisterStore());

```

Just found out that I'm also getting error for dynamic import so it really looks to me it's a configuration problem. I also tried with 
```
...compat.env({
        es2020: true,
    }),
```

but nothing changed. 

But you saying that the default parser should work is already huge help! I'll try to figure out what's going wrong.

**Comment by proko** [comment, marked as answer]
Figured it out. It was because of extending the cypress plugin at the end of the config 🤦🏼 . Moving it at the beginning  fixed the merge order. Thanks again @fasttime
